pkg/util/status.go

```diff
@@ -15,8 +15,10 @@ const (
 	JobRunningReason = "JobRunning"
 	// JobFailedReason is added in a job when it is failed.
 	JobFailedReason = "JobFailed"
-	// JobRestarting is added in a job when it is restarting.
+	// JobRestartingReason is added in a job when it is restarting.
 	JobRestartingReason = "JobRestarting"
+	// JobFailedValidationReason is added in a job when it failed validation
+	JobFailedValidationReason = "JobFailedValidation"
```

Comment:
Do we need this? Why not use `JobFailedReason` and put the right message?

Reply:
It depends on whether we set job to failed when we failed validation.
By the way, the state transition table of job status is not clear now.